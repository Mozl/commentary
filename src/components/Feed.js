import React from 'react';
import styled from 'styled-components';
import data from '../data';

const FeedWrapper = styled.ul`
  display: flex;
  flex-direction: column;
  border: black solid 1px;
`;

FeedWrapper.displayName = 'FeedWrapper';

class Feed extends React.Component {
  render() {
    return (
      <FeedWrapper>
        {data.reverse().map(item => (
          <li key={item.id}>
            {item.body} - {item.id}
          </li>
        ))}
      </FeedWrapper>
    );
  }
}
export default Feed;
