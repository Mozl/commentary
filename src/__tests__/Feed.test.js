import React from 'react';
import ReactDOM from 'react-dom';
import Feed from '../components/Feed';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Feed />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
  const tree = renderer.create(<Feed />);
  expect(tree).toMatchSnapshot();
});

it('renders the list of comments', () => {
  const wrapper = mount(<Feed />);
  expect(wrapper.find('li').length).toEqual(4);
});
