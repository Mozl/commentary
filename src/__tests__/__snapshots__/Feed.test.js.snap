// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  className="sc-bdVaJa kiuZdy"
>
  <li>
    ndsjfsdkjfbkjdsfb
     - 
    0
  </li>
  <li>
    sdfkhksdjfhf
     - 
    1
  </li>
  <li>
    fhdskhfsdfhjsdhfjksdhfjsfadk
     - 
    2
  </li>
  <li>
    fjsdfsdkfsdaklfhsdkajfhsdfldsfhiuwreirr
     - 
    3
  </li>
</ul>
`;
